### PR TITLE
Fixed demos.C redefinition error of dirName

### PR DESCRIPTION
**Pull request by AliAyub007**
Fixed Issue #9764

**Before:**
redifinition-1

**After:**
screenshot 2018-11-07 at 9 12 28 pm

### CODE REVIEW

File: tutorials/demos.C [hunk 1]
```diff
@@ -7,7 +7,7 @@
 ///
 /// \author Rene Brun
 
-{
+void demos() {
    gROOT->Reset();
```
Comment (Axel-Naumann) [inline tutorials/demos.C]: Could you also remove the call to `gROOT->Reset()`. please? It must not show up in named macros (and we don't want to give wrong examples in our tutorials).

**Comment by AliAyub007** [comment]
@Axel-Naumann I also found a very minor issue so I was thinking should I add it to this same PR? as I guess it won't be useful to open another PR for this. 

.? and .help should show same output but there is a bit of difference between them as .? is not being detected from code below: 

screenshot 2018-11-08 at 9 29 34 pm

Just need to update this code by adding a "." and changing 1->2. Thanks!

**Comment by Axel-Naumann** [comment]
Please put it into a separate PR!

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

**Comment by Axel-Naumann** [comment]
Thanks, @AliAyub007 !

**Pull request merged**